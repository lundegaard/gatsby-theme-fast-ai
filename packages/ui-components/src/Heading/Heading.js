import React from 'react';
import PropTypes from 'prop-types';
import { Heading as RebassHeading } from 'rebass';
import { T, always, cond, equals } from 'ramda';
import { alwaysEmptyObject } from 'ramda-extension';

const resolveDefaultProps = cond([
	[equals('h1'), always({ fontSize: [7, 8, 9], mb: [5, 6] })],
	[equals('h2'), always({ fontSize: [3, 4, 7], mb: 3, mt: [2, 3] })],
	[equals('h3'), always({ fontSize: [4, 5, 6], mb: 2, mt: [2, 3] })],
	[equals('h4'), always({ fontSize: [4, 5, 6], mb: 1 })],
	[T, alwaysEmptyObject],
]);

const Heading = ({ as = 'h1', ...rest }) => (
	<RebassHeading
		as={as}
		fontFamily="heading"
		fontWeight="800"
		{...resolveDefaultProps(as)}
		{...rest}
	/>
);

Heading.propTypes = {
	as: PropTypes.string,
};

export default Heading;
