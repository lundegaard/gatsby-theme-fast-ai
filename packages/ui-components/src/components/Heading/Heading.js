import React from 'react';
import PropTypes from 'prop-types';
import { Heading as RebassHeading } from 'rebass';
import { T, always, cond, equals } from 'ramda';
import { alwaysEmptyObject } from 'ramda-extension';

const resolveDefaultProps = cond([
	[equals('h1'), always({ fontSize: [8, 9], mb: [5, 6] })],
	[equals('h2'), always({ fontSize: [4, 7], mb: 3, mt: [4, 5] })],
	[equals('h3'), always({ fontSize: [5, 6], mb: 2 })],
	[T, alwaysEmptyObject],
]);

const Heading = ({ as = 'h1', ...rest }) => (
	<RebassHeading as={as} fontFamily="heading" {...resolveDefaultProps(as)} {...rest} />
);

Heading.propTypes = {
	as: PropTypes.string,
};

export default Heading;
