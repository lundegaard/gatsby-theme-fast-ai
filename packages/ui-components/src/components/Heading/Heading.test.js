import React from 'react';
import { shallow } from 'enzyme';

import Heading from './Heading';

describe('<Heading />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<Heading {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
