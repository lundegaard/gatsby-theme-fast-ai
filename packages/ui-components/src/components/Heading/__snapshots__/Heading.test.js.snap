// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> matches snapshot when shallowly rendered 1`] = `<ForwardRef />`;
